src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(/[;:]/)
+    .map((trimmedString) => trimmedString.trim())
+    .filter((nonEmptyString) => nonEmptyString.length > 0)
+    .reduce((resultObject, currentString, stringIndex, allStrings) => {
+      if (stringIndex % 2 === 0) {
+        resultObject[currentString] = allStrings[stringIndex + 1];
+      }
+
+      return resultObject;
+    }, {});
```

Comment:
[CODE KNOWLEDGE] - It's not a good practice to modify the `resultObject` directly inside the `reduce` method. The `reduce` method should be pure and not cause side effects. Instead, you can return a new object that includes all properties of the `resultObject` and the new property.